com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -242,60 +244,128 @@ public void NetworkSerialize(NetworkSerializer serializer)
+        /// </summary>
+        /// <param name="transformToCommit"></param>
+        /// <param name="dirtyTime"></param>
+        protected void TryCommitTransformToServer(Transform transformToCommit, double dirtyTime)
+        {
+            var isDirty = ApplyTransformToNetworkState(ref m_LocalAuthoritativeNetworkState, dirtyTime, transformToCommit);
 
-        public void ResetCurrentInterpolatedState()
+            void Send()
+            {
+                if (IsServer)
+                {
+                    // server RPC takes a few frames to execute server side, we want this to execute immediately
+                    CommitLocallyAndReplicate(m_LocalAuthoritativeNetworkState);
+                }
+                else
+                {
+                    CommitTransformServerRpc(m_LocalAuthoritativeNetworkState);
+                }
+            }
+
+            // if dirty, send
+            // if not dirty anymore, but hasn't sent last value for limiting extrapolation, still set isDirty
+            // if not dirty and has already sent last value, don't do anything
+            // extrapolation works by using last two values. if it doesn't receive anything anymore, it'll continue to extrapolate.
+            // This is great in case there's message loss, not so great if we just don't have new values to send.
+            // the following will send one last "copied" value so unclamped interpolation tries to extrapolate between two identical values, effectively
+            // making it immobile.
+            if (isDirty)
+            {
+                Send();
+                m_HasSentLastValue = false;
+                m_LastSentTick = NetworkManager.LocalTime.Tick;
+            }
+            else if (!m_HasSentLastValue && NetworkManager.LocalTime.Tick >= m_LastSentTick + 1) // check for state.IsDirty since update can happen more than once per tick. No need for client, RPCs will just queue up
+            {
+                m_LocalAuthoritativeNetworkState.SentTime = NetworkManager.LocalTime.Time; // time 1+ tick later
+                Send();
+                m_HasSentLastValue = true;
+            }
+        }
+
+        [ServerRpc(RequireOwnership = false)]
+        private void CommitTransformServerRpc(NetworkTransformState networkState, ServerRpcParams serverParams = default)
+        {
+            if (serverParams.Receive.SenderClientId == OwnerClientId) // RPC call when not authorized to write could happen during the RTT interval during which a server's ownership change hasn't reached the client yet
+            {
+                CommitLocallyAndReplicate(networkState);
+            }
+        }
+
+        private void CommitLocallyAndReplicate(NetworkTransformState networkState)
+        {
+            m_LocalAuthoritativeNetworkState = networkState;
+            m_ReplicatedNetworkState.Value = networkState;
+            m_ReplicatedNetworkState.SetDirty(true);
```

Comment:
Trying to wrap my head around why I'm dirty now

Reply:
good point. I think this was a remnant of when NetworkState was a class, not a struct. Now it's a struct, SetDirty shouldn't be needed anymore, will remove